### ISSUE TITLE
[BUG]: uncaught connections errors

### ISSUE THREAD

**Issue by HeIIow2** [wasi-master/sponsorblock.py#13]
### Description of the bug

Connection errors should be caught and processed, or at least there should be a `silent` paramater, that says, if an error should be thrown. There is an argument to be made, that this actually shouldn't be the case, but imo the `silent` parameter should exist.

I also reccomend using [logging](https://docs.python.org/3/library/logging.html), then you could just print a warning. When I find the time I am gonna open a pr for this, if you haven't implemented it.

Also sorry if I come of rude, this library really is helpful, but it just crashed my 10 000 lines program. xD
Seriosly keep going. 

<33

**EDIT:** IMO the `silent` attribute should just return an empty array if anything fails, because yk not segments.

### Steps To Reproduce

1. Use the programm
2. Pull out your ethernet cable

**Comment by wasi-master** [comment]
I agree that connection errors should be caught. I don’t know why I didn’t add this the first time lol. But yeah It should be raised as an error.

The `silent` attribute is a good idea too. And I have thought about using `logging` since the beginning but just didn’t get the chance to implement it

I may be able to add these in the next few days

**Comment by HeIIow2** [comment]
Great thank you! You'd do me a huge favor with this.

**Comment by wasi-master** [comment]
Fixed in 08d0c29.

- Connection failures (no internet, timeouts, DNS errors) are now caught for every API call and re-raised as `sponsorblock.errors.ConnectionException`, so you can catch a library exception instead of a raw `requests` one. The original `requests` exception is available on the `.original` attribute.
- There is now a `silent` parameter on the client, and as suggested it returns an empty list instead of raising: with `Client(silent=True)`, `get_skip_segments` and `get_skip_segments_with_hash` log a warning and return `[]` when anything fails (connection error, HTTP error, or invalid response).
- The library now uses `logging` (logger name `sponsorblock.client`), so the warnings can be configured or suppressed like any other logger.

```python
import sponsorblock as sb

client = sb.Client(silent=True)
segments = client.get_skip_segments("kJQP7kiw5Fk")  # [] on failure, never raises
```

Closing as completed.
